taskvine/src/manager/taskvine.h

```diff
@@ -313,6 +313,15 @@ If unset, the library will runs as many functions as it has cores available.
 */
 void vine_task_set_function_slots(struct vine_task *t, int nslots);
 
+/** Set the execution mode of functions inside a library.
+A mode can either be "fork" where the library forks and executes functions, or
+"direct" where the library executes a function in its memory space.
+Note that "direct" will limit the number of function slots in a library to 1.
+@param t                A library object.
+@param exec_mode        A string denoting the execution mode of the library.
```

Comment:
As a general rule, a parameter with a discrete number of useful values should be an enumeration, not a string.  `vine_function_exec_mode_t` or something like that.  This makes it much harder to misuse (or mistype) the value.

Reply:
Actually, I don't think this value is even being sent to the worker as a task property.  (It would have to be explicitly mentioned in `vine_manager_put_task`). Is the library already getting this mode information from `create_library_from_functions` ?

Reply:
So perhaps `t->func_exec_mode` can be dropped entirely?

Reply:
I'm using it so that the manager can enforce that a library with `direct` exec_mode can only have 1 function slot in taskvine/src/manager/vine_manager.c:2731.